hedera-node/hedera-evm-api/src/main/java/com/hedera/services/evm/contracts/operations/HederaEvmSLoadOperation.java

```diff
@@ -0,0 +1,86 @@
+import org.hyperledger.besu.evm.account.Account;
+import org.hyperledger.besu.evm.frame.ExceptionalHaltReason;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.gascalculator.GasCalculator;
+import org.hyperledger.besu.evm.internal.FixedStack;
+import org.hyperledger.besu.evm.operation.AbstractOperation;
+
+public class HederaEvmSLoadOperation extends AbstractOperation {
+    protected final OptionalLong warmCost;
+    protected final OptionalLong coldCost;
+
+    protected final OperationResult warmSuccess;
+    protected final OperationResult coldSuccess;
+
+    protected HederaEvmSLoadOperation(final GasCalculator gasCalculator) {
+        super(0x54, "SLOAD", 1, 1, 1, gasCalculator);
+        final long baseCost = gasCalculator.getSloadOperationGasCost();
+        warmCost = OptionalLong.of(baseCost + gasCalculator.getWarmStorageReadCost());
+        coldCost = OptionalLong.of(baseCost + gasCalculator.getColdSloadCost());
+
+        warmSuccess = new OperationResult(warmCost, Optional.empty());
+        coldSuccess = new OperationResult(coldCost, Optional.empty());
+    }
+
+    @Override
+    public OperationResult execute(MessageFrame frame, EVM evm) {
+        try {
+            final var addressOrAlias = frame.getRecipientAddress();
+            final var worldUpdater = (AbstractLedgerEvmWorldUpdater) frame.getWorldUpdater();
+            final Account account = worldUpdater.get(addressOrAlias);
+            final Address address = account.getAddress();
+            final Bytes32 key = UInt256.fromBytes(frame.popStackItem());
+            final boolean slotIsWarm = frame.warmUpStorage(address, key);
+            final OptionalLong optionalCost = slotIsWarm ? warmCost : coldCost;
+            if (frame.getRemainingGas() < optionalCost.orElse(0L)) {
+                return new OperationResult(
+                        optionalCost, Optional.of(ExceptionalHaltReason.INSUFFICIENT_GAS));
+            } else {
+                UInt256 storageValue = account.getStorageValue(UInt256.fromBytes(key));
+
+                sLoadOPSpecificLogic(frame, address, key, storageValue);
+
+                frame.pushStackItem(storageValue);
+                return slotIsWarm ? warmSuccess : coldSuccess;
+            }
+        } catch (final FixedStack.UnderflowException ufe) {
+            return new OperationResult(
+                    warmCost, Optional.of(ExceptionalHaltReason.INSUFFICIENT_STACK_ITEMS));
+        } catch (final FixedStack.OverflowException ofe) {
+            return new OperationResult(
+                    warmCost, Optional.of(ExceptionalHaltReason.TOO_MANY_STACK_ITEMS));
+        }
+    }
+
+    /**
+     * A placeholder logic method to be overridden by {@link HederaEvmSLoadOperation} descendants.
+     */
+    protected void sLoadOPSpecificLogic(
+            MessageFrame frame, Address address, Bytes32 key, UInt256 storageValue) {
+        // The method is an intentionally-blank
```

Comment:
`The method is an intentionally-blank` -> `The method is intentionally-blank`

Probably we could also rename `sLoadOPSpecificLogic` to `intermediateCustomLogic`. Since the method is inside `HederaEvmSLoadOperation` it's clear that the logic is connected to `SLOAD` operation. I believe `intermediateCustomLogic` better describes the intention of this method.